src/openssl_auth.cpp

```diff
@@ -330,10 +328,15 @@ unsigned char* s3fs_sha256hexsum(int fd, off_t start, ssize_t size)
       DPRNNN("file read error(%d)", errno);
       return NULL;
     }
+
+    const EVP_MD* md        = EVP_get_digestbyname("sha256");
+    sha256ctx = EVP_MD_CTX_create();
+    EVP_DigestInit_ex(sha256ctx, md, NULL);
```

Comment:
You need to declare `sha256ctx` outside the for loop since it accumulates results in every iteration.